requirements-dev.txt

```diff
@@ -20,7 +20,7 @@ seaborn>=0.9.0
 reno>=3.4.0
 Sphinx>=3.0.0
 qiskit-sphinx-theme>=1.6
-sphinx-autodoc-typehints<1.14.0
+sphinx-autodoc-typehints<1.17.1
```

Comment:
Perhaps I should remove this since this is not relevant to the PR (I needed to change this to local document generation). I'm not sure why requirements is written like this. @mtreinish

Reply:
I don't remember why we had the version here pinned. Let's push up a separate pr that removes the version constraint and see what happens. We can debug the docs build issues in parallel to this pr.

Reply:
Undone in https://github.com/Qiskit/qiskit-terra/pull/7821/commits/d4e8c430e3201e07f622f2a24039ce2bdc1766ca